lang/funcs/crypto.go

```diff
@@ -32,6 +33,37 @@ var UUIDFunc = function.New(&function.Spec{
 	},
 })
 
+var UUIDV5Func = function.New(&function.Spec{
+	Params: []function.Parameter{
+		{
+			Name: "namespace",
+			Type: cty.String,
+		},
+		{
+			Name: "name",
+			Type: cty.String,
+		},
+	},
+	Type: function.StaticReturnType(cty.String),
+	Impl: func(args []cty.Value, retType cty.Type) (ret cty.Value, err error) {
+		var namespace uuidv5.UUID
+		switch {
+		case args[0].AsString() == "dns":
+			namespace = uuidv5.NamespaceDNS
+		case args[0].AsString() == "url":
+			namespace = uuidv5.NamespaceURL
+		case args[0].AsString() == "oid":
+			namespace = uuidv5.NamespaceOID
+		case args[0].AsString() == "x500":
+			namespace = uuidv5.NamespaceX500
+		default:
+			return cty.UnknownVal(cty.String), fmt.Errorf("uuidv5() doesn't support namespace %s", args[0].AsString())
```

Comment:
IIRC the namespace portion of a v5 UUID is itself a UUID, with the four you handled above as well-known ones. With that in mind, perhaps we should also allow `namespace` to be given as a literal uuid (requiring the same format that `uuid` and `uuidv5` would return) and use it verbatim if so? That way anyone who is working with a non-standard namespace they allocated themselves (or with a future standard namespace Terraform doesn't support yet) could still use it with this function.

I do like the idea of having short, readable aliases for the standard ones above, though: makes the intent much clearer to the reader in those cases than having the first argument be a literal uuid.

Reply:
I like this little extension and will looking into it. Thanks for Your input.